modular/blocksyncer/database/bucket.go

```diff
@@ -36,6 +36,11 @@ func (db *DB) UpdateBucketToSQL(ctx context.Context, bucket *models.Bucket) (str
 	return stat.SQL.String(), stat.Vars
 }
 
+func (db *DB) UpdateBucketByNameToSQL(ctx context.Context, bucket *models.Bucket) (string, []interface{}) {
+	stat := db.Db.Session(&gorm.Session{DryRun: true}).Table((&models.Bucket{}).TableName()).Where("bucket_name= ?", bucket.BucketName).Updates(bucket).Statement
```

Comment:
There might be several buckets that have same name, with the older ones marked as removed=true. So here we need to find the last bucket with name matches.